### ISSUE TITLE
Investigate compacting the LOH at the start/end of loading

### ISSUE THREAD

**Issue by pchote** [OpenRA/OpenRA#12131]
#12125 and friends appear to be caused by address space exhaustion between loading one giant pile of data and disposing the last.  Compacting the large object heap may help mitigate this.

https://msdn.microsoft.com/en-us/library/system.runtime.gcsettings.largeobjectheapcompactionmode(v=vs.110).aspx

```
GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
GC.Collect();
```

**Comment by GraionDilach** [comment]
Compacting should also happen after a set amount of music switches as well, #11489 is caused by long runner games + the audio loader being inefficient which start/end of loading might cannot solve.

**Comment by LipkeGu** [comment]
When we use this Setting then we enforce .net framework 4.5.1

**Comment by LipkeGu** [comment]
closes #12125

**Comment by LipkeGu** [comment]
i closed #12125 because this Issue is mainly continued. and the reason

**Comment by pchote** [comment]
This is a separate issue to setting the large address aware flag, please don't mix the two.

**Comment by abcdefg30** [comment]
Closing as resolved by #18709.
